containers/daap-python-base/src/var/task/data_product_metadata.py

```diff
@@ -1,116 +1,291 @@
+    "DatabaseName": "",  # add to this from data product name pulled from api path
+    "TableInput": {
+        "Name": "",  # add to this from table name pulled from api path
+        "Description": "",
+        "Owner": "owner",
+        # "Retention": 0,
+        "StorageDescriptor": {
+            "Columns": [],  # add to this from schema passed by user in api request body
+            "Location": "",  # noqa E501 add to this once data land and timestamp generated, although could be top level bucket/data/product/version/table
+            "InputFormat": "org.apache.hadoop.mapred.TextInputFormat",
+            "OutputFormat": "org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat",
+            "Compressed": False,
+            "NumberOfBuckets": -1,
+            "SerdeInfo": {
+                "SerializationLibrary": "org.apache.hadoop.hive.serde2.OpenCSVSerde",
+                "Parameters": {"field.delim": ",", "escape.delim": "\\"},
+            },
+            "BucketColumns": [],
+            "SortColumns": [],
+            "Parameters": {},
+            "StoredAsSubDirectories": False,
+        },
+        "PartitionKeys": [],
+        "TableType": "EXTERNAL_TABLE",
+        "Parameters": {"classification": "csv", "skip.header.line.count": "1"},
+    },
+}
 
-def get_data_product_metadata_spec_path(version: str = "") -> str:
-    # if version is empty we'll get the latest version
-    if version == "":
-        file_paths = get_filepaths_from_s3_folder(
-            DataProductConfig.metadata_spec_prefix().uri
-        )
+
+def get_data_product_specification_path(
+    spec_type: JsonSchemaName, version: None | str = None
+) -> str:
+    """gets path for given version or latest version of metadata and schema json schema"""
+
+    # if version is empty we get the latest version
+    if version is None:
+        file_paths = get_filepaths_from_s3_folder(specification_prefix(spec_type).uri)
         versions = list(
             {i for p in file_paths for i in p.split("/") if i.startswith("v")}
         )
         versions.sort(key=lambda x: [int(y.replace("v", "")) for y in x.split(".")])
         latest_version = versions[-1]
-        path = DataProductConfig.metadata_spec_path(latest_version)
+        path = specification_path(spec_type, latest_version)
     else:
-        DataProductConfig.metadata_spec_path(version)
-        path = DataProductConfig.metadata_spec_path(version)
+        path = specification_path(spec_type, version)
 
     return path.uri
 
 
-def split_bucket_and_key(path):
-    bucket, key = path.replace("s3://", "").split("/", 1)
-    return bucket, key
-
-
-class DataProductMetadata:
-    """
-    class to handle creation and updating of
-    metadata relating to a dataproduct
-    """
+class DataProductBaseJsonSchema:
+    """base class for operations on json type metadata and schema for data products"""
 
-    def __init__(self, data_product_name: str, logger: DataPlatformLogger):
+    def __init__(
```

Comment:
I think this abstraction of `DataProductMetadata` is a good idea, it feels like this class extracts all the stuff that relates a working copy of some json to it's persisted version in s3. It doesn't really have much to do with data products at all any more.

If inheriting from this class is too awkward, another option is to use composition, where the other 2 class instantiate this class directly, and then delegate method calls to it. I think either approach will work here though.